idaes/models_extra/power_generation/unit_models/heat_exchanger_common.py

```diff
@@ -0,0 +1,781 @@
+        @blk.Constraint(
+            blk.flowsheet().config.time,
+            shell.length_domain,
+            doc="velocity on shell side",
+        )
+        def v_shell_eqn(b, t, x):
+            return (
+                b.v_shell[t, x]
+                * shell.properties[t, x].dens_mol_phase["Vap"]
+                * b.area_flow_shell_min
+                == shell.properties[t, x].flow_mol
+            )
+
+        # Reynolds number
+        @blk.Constraint(
+            blk.flowsheet().config.time,
+            shell.length_domain,
+            doc="Reynolds number equation on shell side",
+        )
+        def N_Re_shell_eqn(b, t, x):
+            return (
+                b.N_Re_shell[t, x] * shell.properties[t, x].visc_d_phase["Vap"]
+                == b.do_tube
+                * b.v_shell[t, x]
+                * shell.properties[t, x].dens_mol_phase["Vap"]
+                * shell.properties[t, x].mw
+            )
+
+    if shell_has_pressure_change:
+        # Friction factor on shell side
+        if blk.config.tube_arrangement == "in-line":
+
+            @blk.Constraint(
+                blk.flowsheet().config.time,
+                shell.length_domain,
+                doc="in-line friction factor on shell side",
+            )
+            def friction_factor_shell_eqn(b, t, x):
+                return (
+                    b.friction_factor_shell[t, x] * b.N_Re_shell[t, x] ** 0.15
+                    == (
+                        0.044
+                        + 0.08
+                        * b.pitch_x_to_do
+                        / (b.pitch_y_to_do - 1.0) ** (0.43 + 1.13 / b.pitch_x_to_do)
+                    )
+                    * b.fcorrection_dp_shell
+                )
+
+        elif blk.config.tube_arrangement == "staggered":
+
+            @blk.Constraint(
+                blk.flowsheet().config.time,
+                shell.length_domain,
+                doc="staggered friction factor on shell side",
+            )
+            def friction_factor_shell_eqn(b, t, x):
+                return (
+                    b.friction_factor_shell[t, x] * b.N_Re_shell[t, x] ** 0.16
+                    == (0.25 + 0.118 / (b.pitch_y_to_do - 1.0) ** 1.08)
+                    * b.fcorrection_dp_shell
+                )
+
+        else:
+            raise ConfigurationError()
+
+        # Pressure drop on shell side
+        @blk.Constraint(
+            blk.flowsheet().config.time,
+            shell.length_domain,
+            doc="pressure change on shell side",
+        )
+        def deltaP_shell_eqn(b, t, x):
+            # FIXME this equation doesn't have elevation change---is this right?
+            return (
+                b.deltaP_shell[t, x] * b.pitch_x
+                == -1.4
+                * b.friction_factor_shell[t, x]
+                * shell.properties[t, x].dens_mol_phase["Vap"]
+                * shell.properties[t, x].mw
+                * b.v_shell[t, x] ** 2
+            )
```

Comment:
@jinliangma , should this equation have an elevation change term?

Reply:
Original cross flow heat exchanger is for boiler superheaters and economizers. The shell side has flue gas that is very low in density such that the static pressure difference due to elevation change can be neglected.  We only considered the tube side elevation effect since economizer's tube side has liquid water.  I believe this model now considers only the vapor phase so we may remove the elevation term for both sides.  It won't hurt if we add those terms for both sides though.